### PR TITLE
PP-12175 upgrade Dropwizard to version 3

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID

- Upgrade Dropwizard to v3 latest.
- Use dropwizard-dependencies BOM, which removes the need to pull in some dependencies ourselves

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/resources/PaymentsResource.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
 import uk.gov.pay.api.service.PaymentSearchService;
 import uk.gov.pay.api.service.PublicApiUriGenerator;
 
-import javax.annotation.Nullable;
+import org.checkerframework.checker.nullness.qual.Nullable;
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/api/resources/PaymentsResource.java:35]: 🤨

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Looks great apart from one import…

File: pom.xml [hunk 2]
```diff
@@ -130,62 +178,91 @@
...
-        <!-- testing -->
         <dependency>
-            <groupId>uk.gov.service.payments</groupId>
-            <artifactId>testing</artifactId>
-            <version>${pay-java-commons.version}</version>
-            <scope>test</scope>
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
+            <version>1.3.2</version>
+        </dependency>
```
Comment (alexbishop1) [inline pom.xml:195]: Do we need this and the FindBugs JSR 305 dependency?

File: pom.xml [hunk 3]
```diff
@@ -130,62 +178,91 @@
...
+        </dependency><dependency>
+        <groupId>com.google.code.gson</groupId>
+        <artifactId>gson</artifactId>
+        <version>2.10.1</version>
+    </dependency>
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
+            <version>3.0.2</version>
+        </dependency>
```
Comment (alexbishop1) [inline pom.xml:220]: Double paste?

**Review by alexbishop1 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (SandorArpa) [reply pom.xml:220]: 🤦

File: pom.xml (hunk 2 above)
Comment (SandorArpa) [reply pom.xml:195]: No

**Review by alexbishop1 — APPROVED** [review]
🚀

**Pull request merged**